scripts/contract_matcher.js

```diff
@@ -0,0 +1,85 @@
+const bytecodeUtils = require('bytecode-utils');
+const fs = require('fs');
+const url = require('url');
+const path = require('path');
+const truffleFlattener = require('truffle-flattener');
+
+module.exports = async (consulConnectionURL, chainUrl, chainName, polySidechainName, headers = {}) => {
+	const consulBaseUrl = `chain/${polySidechainName}`;
+	const consulUrl = new url.parse(consulConnectionURL);
+	const consul = require('consul')({ host: consulUrl.hostname, port: consulUrl.port, promisify: fromCallback, headers });
+	let response;
+
+	try {
+		response = await consul.kv.get(`${consulBaseUrl}/${chainName}`);
+	} catch (e) {
+		console.log(e);
+		console.error(`Failed to connect to consul at ${consulBaseUrl}${configPath}`)
+		process.exit(1);
+	}
+
+	const [chainConfig, resHeaders] = response;
+
+	if (resHeaders.statusCode == 500) {
+		console.error('There was an internal consul error, bailing.');
+		process.exit(1);
+	} else if (resHeaders.statusCode == 404) {
+		console.log('Didn\'t find consul config, proceeding.');
+		console.log('Recieved status code error: ' + resHeaders.statusCode);
+		return true;
+	}
+
+	const config = JSON.parse(chainConfig.Value);
+
+	console.log(`Checking for difference in contracts on ${chainName}`);
+
+	let results = await Promise.all([doesMatchExist(chainUrl, 'NectarToken', config.nectar_token_address),
+		doesMatchExist(chainUrl, 'BountyRegistry', config.bounty_registry_address),
+		doesMatchExist(chainUrl, 'OfferRegistry', config.offer_registry_address),
+		doesMatchExist(chainUrl, 'ERC20Relay', config.erc20_relay_address)
+	]);
+
+	if (results.some(isMatchingBytecode => !isMatchingBytecode)) {
```

Comment:
It is desired to always redeploy all, because otherwise we need dependency tracking logic here and need to maintain it in lockstep with contract deps, which would be fragile.